common/sampling.cpp

```diff
@@ -189,7 +190,11 @@ llama_token llama_sampling_sample(
             llama_sample_top_k    (ctx_main, &cur_p, top_k,     min_keep);
             llama_sample_tail_free(ctx_main, &cur_p, tfs_z,     min_keep);
             llama_sample_typical  (ctx_main, &cur_p, typical_p, min_keep);
-            llama_sample_top_p    (ctx_main, &cur_p, top_p,     min_keep);
+            if (min_p != 0.0) {
+                llama_sample_min_p(ctx_main, &cur_p, min_p,     min_keep);
+            } else {
+                llama_sample_top_p(ctx_main, &cur_p, top_p,     min_keep);
+            }
```

Comment:
Question: Why do these have to be mutually exclusive? Perhaps min_p is superior, but I don't see how using one would prevent you from using the other, in either order.

Reply:
Yes, it can be just:
```
llama_sample_min_p(ctx_main, &cur_p, min_p,     min_keep);
llama_sample_top_p(ctx_main, &cur_p, top_p,     min_keep);
```

Reply:
Technically, they don't have to be separated, but I wanted to avoid stacking samplers like this for consistency's sake. There will probably be people specifying Min P not realizing that Top P is on by default as well, when it was designed more or less as a replacement.
However if it makes more sense to you to leave this fully up to the user I understand that approach as well, I just think stacking samplers with different design strategies can complicate things more than necessary.

Reply:
The more I think about it... it does make sense that it's not something that the software should artficially impose on you. I think we can remove the exclusivity, at least for the time being.